modules/lazyload/module.php

```diff
@@ -64,21 +65,46 @@ private function update_element_attributes( Element_Base $element ) {
 	}
 
 	private function append_lazyload_selector( $control, $value ) {
+		$is_dominant_color_enabled = $this->is_dominant_color_enabled();
+
 		if ( Utils::get_array_value_by_keys( $control, [ 'background_lazyload', 'active' ] ) ) {
 			foreach ( $control['selectors'] as $selector => $css_property ) {
 				if ( 0 === strpos( $css_property, 'background-image' ) ) {
 					if ( ! empty( $value['url'] ) ) {
 						$css_property  = str_replace( 'url("{{URL}}")', 'var(--e-bg-lazyload-loaded)', $css_property );
 						$control['selectors'][ $selector ] = $css_property . '--e-bg-lazyload: url("' . $value['url'] . '");';
-						$control = $this->apply_dominant_color_background( $control, $value, $selector );
+
+						if ( $is_dominant_color_enabled ) {
+							$control = $this->apply_dominant_color_background( $control, $value, $selector );
+						}
 					}
 				}
 			}
 		}
 		return $control;
 	}
 
+	private function is_dominant_color_enabled() {
+		$performance_lab_settings = (array) get_option( self::PERFORMANCE_LAB_OPTION_NAME, [] );
```

Comment:
Querying the database for the relevant option is the only way to find out if the option is enabled. As performance_lab is also doing this by themselves.

There is no need to check if the plugin is activated, or if any event files exist by trying to use some namespace. This will only indicate that the plugin exists in general, but not the value of the option.

Could you think of a way to get the option value without using the DB?